_partials/_migrate_live_migrate_faq_all.md

```diff
@@ -0,0 +1,103 @@
+migrates roles without passwords. You have to migrate passwords manually.
+
+
+### Table privileges
+
+Live-migration does not migrate table privileges. After completing Live-migration: 
+
+1. Grant all roles to `tsdbadmin`.
+   ```shell
+   psql -d $SOURCE -t -A -c "SELECT FORMAT('GRANT %I TO tsdbadmin;', rolname) FROM 
+   pg_catalog.pg_roles WHERE rolname not like 'pg_%' AND rolname != 'tsdbadmin' 
+   AND NOT rolsuper" | psql -d $TARGET -f -  
+   ```
+   
+1. On your migration machine, edit `/tmp/grants.psql` to match table privileges on your source database.
+   ```shell
+   pg_dump --schema-only --quote-all-identifiers 
+   --exclude-schema=_timescaledb_catalog --format=plain --dbname "$SOURCE" | grep 
+   "(ALTER.*OWNER.*|GRANT|REVOKE)"  > /tmp/grants.psql 
+   ```
+   
+1. Run `grants.psql` on your target Timescale Cloud service. 
+   ```shell
+   psql -d $TARGET -f /tmp/grants.psql
+   ```
+
+### Postgres to Timescale: “live-replay not keeping up with source load”
+
+1. Go to the Timescale cloud console -> Insights tab and find the query which takes significant time
+2. If the query is either UPDATE/DELETE, make sure the columns used on the WHERE clause have necessary indexes.
+3. If the query is either UPDATE/DELETE on the tables which are converted as hypertables, make sure the REPLIDA IDENTITY(defaults to primary key) on the source is compatible with the target primary key. If not, create an UNIQUE index source database by including the hypertable partition column and make it as a REPLICA IDENTITY. Also, create the same UNIQUE index on target.
+
+### ERROR:  invalid snapshot identifier: "xxxxxx" (or) SSL SYSCALL error: EOF detected on RDS
+
+This rare phenomenon may happen when:
+
+- The snapshot becomes stale due to network connection interruption between the snapshot process and your source database.
+
+  When you see you this error you need to tune the tcp parameter tuning on your source RDS instance. Update the
+  following GUCs to the recommended values on the source RDS instance.
+
+   ```shell
+   psql -X -d $SOURCE -c 'alter system set tcp_keepalives_count=60'
+   psql -X -d $SOURCE -c 'alter system set tcp_keepalives_idle=10'
+   psql -X -d $SOURCE -c 'alter system set tcp_keepalives_interval=10'
+   psql -X -d $SOURCE -c 'alter system set wal_sender_timeout=30m'
+   ```
+
+  For more information, see [https://github.com/dimitri/pgcopydb/issues/773#issuecomment-2139093365](https://github.com/dimitri/pgcopydb/issues/773#issuecomment-2139093365)
+
+- Either your source database or target Timescale Cloud service is under CPU/Memory/Disk/Network pressure:
+
+  Upgrade to better instances types until migration completes.
+
+### duplicate key value violates the `metadata_pkey` unique constraint on TimescaleDB
+
+Live-migration is compatible with TimescaleDB v2.14.x and greater. To migrate from previous versions,
+[Align the version of TimescaleDB on the source and target][align-versions].
+
```

Comment:
This can be removed. We are handling this automatically from the tool.

Reply:
Done